Added Branch Coverage Tool & Unit Tests for parseEntry()

simonsirak opened the pull request:
<!-- 
Describe the changes you have made here: what, why, ... 
Link issues by using the following pattern: #333.
If you fixed a koppor issue, link it with following pattern: [koppor#47](https://github.com/koppor/jabref/issues/47).
The title of the PR must not reference an issue, because GitHub does not support autolinking there.
-->
#### Added
* Branch coverage tool for `GvkParser::parseEntry()`. The tool considered `if`, `else` (and if one was missing, it was inserted), `for` (and skipped `for` loops). It does not consider nested `if` statements that are specifically a result of short-circuiting. No ternary operators or try/catch were encountered. The tool has results that are relatively consistent with the automated tool for this project, `Jacoco`.
* 4 cases that increase the branch coverage from `~75%` to `~84%` using our coverage tool.

simonsirak committed:
Add branch coverage metric to GvKParser::parseEntry

simonsirak committed:
Add manual coverage results

simonsirak committed:
Add 2 tests for parseEntry

Increases manual branch coverage by ~5 percentage points.

simonsirak committed:
Simplify test input for test030F

This simplifies test inspection in the
future.

simonsirak committed:
Add final test cases

Add final test cases and simplify
the existing test cases. Now, they
are properly based on the PICA spec.

yiiju reviewed (approved):
Looks good 👍

christinerosquist reviewed (approved):
Nice job 👍

SoFoDa reviewed (changes requested):
`parseEntry()` seems to be a good candidate in terms of refactoring. We could refactor each tag procedure into smaller functions before merging.

simonsirak committed:
Move local vars to global private scope

christinerosquist committed:
Started refactoring parseEntry

SoFoDa committed:
refactored first five tags

simonsirak committed:
Refactor parseEntry from 027D.

Refactoring added helper functions with descriptive
names, as well as a separate helper function for
configuring the final BibEntry

christinerosquist committed:
changed function names

christinerosquist committed:
resolved merge conflict

yiiju committed:
Refactor parseEntry from 036E to 037C

simonsirak committed:
Modify branch coverage tool

Now it is adjusted based on the refactoring

simonsirak committed:
Modify helper names

Change names to same structure

simonsirak committed:
Add curly brace
